### DISCUSSION TITLE
[Proposal] Simplification of Constant Declarations.

### DISCUSSION THREAD

**Discussion by killer-frog** [dotnet/csharplang#3478] [Language Ideas]
Often find myself declaring constants where the value isn't actually important - just having the same constant is.  So often i write

`const string LastUpdatedKey = "LastupdatedKey";`

would be nice if we could just write

`const string LastUpdatedKey;`

**Comment by svick** [comment]
This would mean that `const string LastUpdatedKey;` and `readonly string LastUpdatedKey;` are initialized to completely different values. I don't think that would be a good idea.

**Comment by killer-frog** [comment]
interesting point, although the difference is that a `readonly string` is set at runtime whilst a `const string ` is define at compile time (in my suggestion the standard const definition of a string would still be valid)

for a read only string this is valid 

```
	public class Cl
	{
		public readonly string str;
		
		public Cl (string val)
		{
			str = val;
		}
	}
```

**Comment by huoyaoyuan** [comment]
```
const string LastUpdatedKey = nameof(LastUpdatedKey);
```

This pattern is widely used in the roslyn codebase.
